src/PointwiseFunctions/AnalyticSolutions/GrMhd/BondiMichelMHD.hpp

```diff
@@ -0,0 +1,207 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <limits>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "NumericalAlgorithms/Interpolation/BarycentricRational.hpp"
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/Hydro/Tags.hpp"  // IWYU pragma: keep
+#include "Utilities/MakeArray.hpp"            // IWYU pragma: keep
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace grmhd {
+namespace Solutions {
+
+/*!
+ * \brief Bondi-Michel accretion with superposed magnetic field in Schwarzchild
+ * spacetime.
+ *
+ * Modulo the magnetic field, an analytic solution to the 3-D GrMhd system.
+ * The user specifies the sonic radius \f$r_c\f$ and sonic rest mass density
+ * \f$\rho_c\f$, which are the radius and rest mass density at the sonic point,
+ * the radius at which the fluid's eulerian velocity as seen by a distant
+ * observer overtakes its sound speed. These quantities can be related to the
+ * sound speed at infinity. In the case of the interstellar medium, the sound
+ * speed is \f$\approx 10^{-4}\f$, which results in a sonic radius of
+ * \f$\approx 10^8 M\f$ for \f$\Gamma \neq 5/3\f$ (Rezzola and Zanotti, 2013).
+ *
+ * The user also specifies the max radius, which is useful if one does not want
+ * to extend the domain up to the remote sonic point. Additionally specified by
+ * the user are the adiabatic exponent \f$\Gamma\f$, and the magnitude of the
+ * magnetic field \f$B\f$.
+ * In Schwarschild-Cartesian coordinates \f$(x, y, z)\f$, where
+ * \f$ r = \sqrt{x^2 + y^2 + z^2}\f$, the superposed magnetic field is
+ *
+ * \f{align*}
+ * B_x(\vec{x},t) &= \frac{B x}{r^3 \sqrt{1 + 2/r}} \\
+ * B_y(\vec{x},t) &= \frac{B y}{r^3 \sqrt{1 + 2/r}} \\
+ * B_z(\vec{x},t) &= \frac{B z}{r^2 \sqrt{1 + 2/r}}
+ * \f}
```

Comment:
Unfortunately I did not look too deeply into the details of where the magnetic field on the bitbucket code came from. I also believe I might have missed the transformation from isotropic coordinates for the magnetic field quantities, so thanks for catching that! @ffoucart I believe you implemented the bitbucket version; do you recall where this magnetic field comes from? I could add it to the documentation :+1:

Reply:
The magnetic field has to satisfy two conditions:

(1) It is radial, so that it does not modify the steady-state solution for pure hydro accretion
(2) It satisfies div(B)=0.

That leads to the solution

* \f{align*}
 * B^x(\vec{x},t) &= \frac{B0*x}{r^3 \sqrt{1 + 2/r}} \\
 * B^y(\vec{x},t) &= \frac{B0*y}{r^3 \sqrt{1 + 2/r}} \\
 * B^z(\vec{x},t) &= \frac{B0*z}{r^3 \sqrt{1 + 2/r}}
 * \f}

as in lines 257-264 of the bitbucket code, which is slightly different from the one in your comments. B0 is a free parameter setting the strenght of the B-field.
[The metric terms will be coordinate dependent, so this assumes that you use the same setup as in the bitbucket code... otherwise, just make sure that div(B)=0 ]